src/util/file.cc

```diff
@@ -132,6 +132,14 @@ future<stat_data> file_stat(std::string_view name, follow_symlink follow) noexce
     return engine().file_stat(name, follow);
 }
 
+future<group_details> grp_detail(std::string_view groupname) noexcept {
+    return engine().grp_detail(groupname);
+}
+
+future<> change_ownership_of_file(std::string_view filepath, uint64_t groupgid) noexcept {
```

Comment:
s/filepath/pathname/ to be more consistent.

why not using `gid_t` for the type of gid?

i'd suggest rename `groupgid` to `group` , as the "group" in `groupgid` is redundant.

Reply:
Right. <grp.h> already has git_t and struct group (which getgrnam returns), no need to invent our own copies of the same thing.